app/src/org/commcare/CommCareHeartbeatManager.java

```diff
@@ -0,0 +1,287 @@
+package org.commcare;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Handler;
+import android.os.Looper;
+import android.util.Base64;
+
+import org.commcare.activities.PromptUpdateActivity;
+import org.commcare.core.interfaces.HttpResponseProcessor;
+import org.commcare.core.network.ModernHttpRequester;
+import org.commcare.preferences.CommCareServerPreferences;
+import org.commcare.utils.SessionUnavailableException;
+import org.commcare.utils.StorageUtils;
+import org.commcare.utils.SyncDetailCalculations;
+import org.javarosa.core.io.StreamsUtil;
+import org.javarosa.core.model.User;
+import org.javarosa.core.util.externalizable.ExtUtil;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.ByteArrayInputStream;
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Timer;
+import java.util.TimerTask;
+
+/**
+ * Created by amstone326 on 4/13/17.
```

Comment:
Otherwise, seems like this class is still anticipated some changes from HQ until its finished up so will wait until this happens before digging in.